src/components/chainStateBlock/index.tsx

```diff
@@ -1,52 +1,160 @@
+  const chainSpecs = useStoreChain?.use?.chainSpecs?.();
+  const chain = useStoreChain?.use?.chain?.();
+
+  const showData = chainData && chainSpecs;
+  const isParaChain = chain.isParaChain;
+  const chainHasStaking = chain.hasStaking;
+
+  const isStakingBlock = type === 'totalStake';
+  const isBlockTimeBlock = type === 'blockTime';
+
+  let data;
+
+  const formatData = useCallback((value: number | bigint) => {
+    return formatNumber(
+      Number(formatTokenValue({
+        value: Number(value),
+        precision: 2,
+        tokenDecimals: chainSpecs?.properties.tokenDecimals,
+      })),
+    );
+  }, [chainSpecs]);
+
+  switch (type) {
+    case 'totalIssuance':
+      if (showData) {
+        data = formatData(chainData);
+      }
+      break;
+
+    case 'totalStake':
+      if (!chainHasStaking) {
+        data = 'No Staking Data';
+        break;
+      }
+
+      if (showData) {
+        data = formatData(chainData);
+      }
+      break;
+
+    case 'blockTime':
+      if (chainData) {
+        const blockTimeSec = Number(chainData) / 1000;
+        data = blockTimeSec;
+      }
+      break;
+
+    case 'bestBlock':
+    case 'finalizedBlock':
+      if (showData) {
+        data = formatNumber(Number(chainData));
+      }
+      break;
 
-  const chainData = useStoreChain?.use?.[typeLib[type]]?.();
+    default:
+      break;
+  }
 
   return (
     <div className={cn(
-      'grid grid-cols-[16px_1fr] items-center gap-4',
+      'grid grid-cols-[16px_1fr] items-center gap-2',
     )}
     >
-      <Icon name={chainStateBlockData[type].icon} size={[16]} />
+      <Icon name={chainStateBlockData?.[type]?.icon} size={[16]} />
       <div className="flex flex-col overflow-hidden">
         <span className={cn(
           'truncate font-geist font-body2-regular',
           'text-dev-black-300 dark:text-dev-purple-300',
         )}
         >
-          {chainStateBlockData[type].name}
+          {chainStateBlockData?.[type]?.name}
+
+          {isBlockTimeBlock && data && ` ${data} sec`}
+          {isBlockTimeBlock && !data && ` Loading...`}
+
+          {isStakingBlock && isParaChain && `at ${chain.relayChainId}`}
         </span>
         <span className="truncate font-geist font-body1-bold">
           {
-            !chainData
-              ? 'Loading...'
-              : formatNumber(chainData)
+            isBlockTimeBlock
+              ? <BlockTarget target={data as number} />
+              : data || 'Loading...'
           }
         </span>
       </div>
     </div>
   );
 };
+
+const BlockTarget = ({ target }: { target: number }) => {
+  const latestBlock = useStoreChain?.use?.bestBlock?.();
+  const chain = useStoreChain?.use?.chain?.();
+
+  const refBlockDuration = useRef<HTMLSpanElement>(null);
+  const refLastChain = useRef<StoreInterface['chain']['id']>(chain?.id);
+
+  useEffect(() => {
+    let timePassed = 0;
+    let interval: NodeJS.Timeout;
+
+    if (!refBlockDuration?.current?.innerText) {
```

Comment:
textContent is better in this case ... we need to change only the text without adding html tags ;)